pkg/auth/roleTemplate.go

```diff
@@ -63,25 +59,12 @@ func (r *RoleTemplateResolver) gatherRules(roleTemplate *rancherv3.RoleTemplate,
 	seen[roleTemplate.Name] = true
 
 	if roleTemplate.External {
-		externalRulesEnabled, err := r.isExternalRulesFeatureFlagEnabled()
-		if err != nil {
-			return nil, fmt.Errorf("failed to check externalRules feature flag: %w", err)
-		}
-
-		if externalRulesEnabled {
-			if roleTemplate.ExternalRules != nil {
-				rules = append(rules, roleTemplate.ExternalRules...)
-			} else {
-				cr, err := r.clusterRoles.Get(roleTemplate.Name)
-				if err != nil {
-					return nil, fmt.Errorf("for external RoleTemplates, externalRules must be provided or a backing clusterRole must be installed to check for privilege escalations: failed to get ClusterRole %q: %w", roleTemplate.Name, err)
-				}
-				rules = append(rules, cr.Rules...)
-			}
-		} else if roleTemplate.Context == "cluster" {
+		if roleTemplate.ExternalRules != nil {
+			rules = append(rules, roleTemplate.ExternalRules...)
+		} else {
 			cr, err := r.clusterRoles.Get(roleTemplate.Name)
 			if err != nil {
-				return nil, fmt.Errorf("failed to get ClusterRole %q: %w", roleTemplate.Name, err)
+				return nil, fmt.Errorf("for external RoleTemplates, externalRules must be provided or a backing clusterRole must be installed to check for privilege escalations: failed to get ClusterRole %q: %w", roleTemplate.Name, err)
```

Comment:
Explicitly noting that this looks ok, given that a similar change in the https://github.com/rancher/rancher/pull/45864#pullrequestreview-2138671649 got marked as problem.